site/src/pages/DeploySettingsPage/LicensesSettingsPage/AddNewLicensePage.tsx

```diff
@@ -0,0 +1,53 @@
+import { useMutation } from "@tanstack/react-query"
+import { createLicense } from "api/api"
+import { displayError, displaySuccess } from "components/GlobalSnackbar/utils"
+import { FC } from "react"
+import { useNavigate } from "react-router-dom"
+import { AddNewLicensePageView } from "./AddNewLicensePageView"
+import { pageTitle } from "utils/page"
+import { Helmet } from "react-helmet-async"
+
+const AddNewLicensePage: FC = () => {
+  const navigate = useNavigate()
+
+  const {
+    mutate: saveLicenseKeyApi,
+    isLoading: isCreating,
+    isError: creationFailed,
+  } = useMutation(createLicense, {
+    onSuccess: () => {
+      displaySuccess("You have successfully added a license")
+      navigate("/settings/deployment/licenses?success=true")
+    },
+    onError: () => displayError("Failed to save license key"),
+  })
+
+  function saveLicenseKey(licenseKey: string) {
+    saveLicenseKeyApi(
+      { license: licenseKey },
+      {
+        onSuccess: () => {
+          displaySuccess("You have successfully added a license")
+          navigate("/settings/deployment/licenses?success=true")
+        },
+        onError: () => displayError("Failed to save license key"),
```

Comment:
I think it would be nice to have a more explicit error if someone is adding a duplicate license. This is a case we might run into more frequently, given that it is currently not very easy to identify which license is which (they are unnamed).

Reply:
done! Thx!